Data Structures/Queues/circularqueue.java

```diff
@@ -0,0 +1,99 @@
+import java.io.*;
+import java.util.*;
+
+class cqueue
+{
+	int f=-1,r=-1; //here f is front pointer and r is rear pointer.
+	Scanner x=new Scanner(System.in);
+	int queue[]=new int[5]; //maximum size of queue is 5.
+	public void insert()
+	{
+		int e;
+		if((f==0 && r==4)||((r+1)==f))
+			System.out.print("\nThe queue is full");
+		else if (f==-1 && r==-1)
+		{
+			f=0;
+			r++;
+			System.out.print("\nInsert element:");
+			e=x.nextInt();
+			queue[r]=e;
+		}
+		else if(f!=0 && r==4)
+		{
+			r=0;
+			System.out.print("\nInsert element:");
+			e=x.nextInt();
+			queue[r]=e;
+		}
+		else
+		{
+			r++;
+			System.out.print("\nInsert element:");
+			e=x.nextInt();
+			queue[r]=e;
+		}
+	}
+	public void display()
+	{
+		System.out.print("\nThe queue is: ");
+		if((f==0)||(f<r))
+		{
+			for(int i=f;i<=r;i++)
+				System.out.print(" "+queue[i]);
+		}
+		else if(f>r)
+		{
+			for(int k=f;k<=4;k++)
+				System.out.print(" "+queue[k]);
+			for(int l=0;l<=r;l++)
+				System.out.print(" "+queue[l]);
+		}
+		else
+			System.out.println();
+	}
+	public void delet()
+	{
+		if(f==-1 && r==-1)
+			System.out.print("\nThe queue is empty.");
+		else if(queue[f]=='\0' && queue[r]=='\0')
+			System.out.print("\nThe queue is empty.");
+		else if(f>=0)
+		{
+			queue[f]='\0';
```

Comment:
This mark is error-prone. What is if the user give a 0 as input? Use a negative value and inform the user that he can give only positive integers.